### PR TITLE
Chore: add snippets for api section

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the snippets required for the api section are included and the markdown files are now adjusted to reflect the changes:

What was done:

- Updated the markdown files to include the snippets based on the file naming and structure decided.
- Created the snippets.

Feel free to provide feedback

### CODE REVIEW

**Review by domyen — APPROVED** [review]

**Comment by domyen** [comment]
lgtm! 👏

File: docs/snippets/common/badge-story-starter-example.mdx.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+```md
+
+<!--- Badge.stories.mdx --->
+
+<!--- This is your Story template function, shown here in React -->
```
Comment (tmeasday) [inline docs/snippets/common/badge-story-starter-example.mdx.mdx:5]: I think maybe you misunderstood my meaning in including these comments. I was referring to the MDX files where there is a single line:

```mdx
<!--- This is your Story template function, shown here in React -->
export const Template = (args) => <Button {...args} />

<Story name="X" args={{bla}} >
  {Template.bind({})}
</Story>
```

In such cases that line (`export const Template ...`) is the single thing that would be different between a react + angular version of the MDX snippet.

However in many of these examples there are large snippets of react-specific code and so I think they should "react" snippets. (Actually in this actual example I think we should maybe rewrite the snippet to use args, but in other examples they need to remain "react-ized").

File: docs/snippets/common/my-component-story-basic-and-props.js.mdx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// MyComponent.story.js
+import MyComponent from './MyComponent';
+
+export default {
+  title: 'Path/To/MyComponent',
+  component: MyComponent,
+};
+
+export const Basic = () => <MyComponent />;
+export const WithProp = () => <MyComponent prop="value" />;
```
Comment (tmeasday) [inline docs/snippets/common/my-component-story-basic-and-props.js.mdx:11]: Isn't this react code?

File: docs/snippets/react/storybook-addon-change-panel.js.mdx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+```js
+// .storybook/my-addon/register.js
+
+import { useParameter } from '@storybook/api';
+const PARAM_KEY = 'myAddon';
+const MyPanel = () => {
+  const value = useParameter(PARAM_KEY, null);
+  const item = value ? value.data : '';
+  return <div>{item}</div>;
+};
```
Comment (tmeasday) [inline docs/snippets/react/storybook-addon-change-panel.js.mdx:10]: This snippet is common -- not react specific (manager code is always react).

File: docs/snippets/react/storybook-addon-disable-addon.js.mdx [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+// .storybook/my-addon/register.js
+
+addons.register(ADDON_ID, () => {
+  addons.add(PANEL_ID, {
+    type: types.PANEL,
+    title: 'My Addon',
+    render: () => <div>Addon tab content</div>,
+    paramKey: 'myAddon', // this element
+  });
+});
```
Comment (tmeasday) [inline docs/snippets/react/storybook-addon-disable-addon.js.mdx:11]: This snippet is common -- not react specific (manager code is always react).

File: docs/snippets/react/storybook-addon-panel-initial.js.mdx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  addons.add(PANEL_ID, {
+    type: types.PANEL,
+    title: 'My Addon',
+    render: ({ active, key }) => (
+      <AddonPanel active={active} key={key}>
+        <MyPanel />
+      </AddonPanel>
+    ),
+  });
+});
```
Comment (tmeasday) [inline docs/snippets/react/storybook-addon-panel-initial.js.mdx:23]: This snippet is common -- not react specific (manager code is always react).

**Review by tmeasday — CHANGES_REQUESTED** [review]
Looks good, a few minor changes

**Comment by jonniebigodes** [comment]
@tmeasday with the latest commit i addressed the feedback you mentioned. 

Let me know if you need anything else.

**Review by tmeasday — APPROVED** [review]
LGTM

**Comment by jonniebigodes** [comment]
conflicts are solved, let me know if you need anything else.

**Pull request merged**